calyx-py/calyx/py_ast.py

```diff
@@ -5,8 +5,16 @@
 import inspect
 import os
 
-EMIT_FILEINFO = "CALYX_PY_SOURCELOC" in os.environ and bool(
-    os.environ.get("CALYX_PY_SOURCELOC")
+"""
+The global variable "CALYX_PY_SOURCELOC" determines whether, and how, source-location metadata will be emitted.
+
+The modes are
+"" (empty string): Omit sourceloc information
+"test": Output sourceloc information with relative paths (should only be used for testing)
+otherwise, output sourceloc information with absolute paths
```

Comment:
Can we instead add an option to supply a path that all source info should be relative to? Something like `CALYX_PY_SOURCELOC_ROOT`? Then people could just specify the root directory of their project to get portable paths